spec/mailers/reminder_deadline_mailer_spec.rb

```diff
@@ -1,8 +1,15 @@
 describe ReminderDeadlineMailer, type: :job, skip_seed: true do
   describe 'notify deadline' do
+    let(:today) { Date.new(2022, 1, 10) }
+    let(:organization) { create(:organization, skip_items: true, reminder_day: today.day) }
+    let(:partner) { create(:partner, organization: organization) }
+    let(:date) { Date.new(2022, 1, 1) }
+
     subject { described_class.notify_deadline(partner) }
-    let(:partner) { create(:partner) }
-    let(:organization) { partner.organization }
+
+    before do
+      allow(DeadlineService).to receive(:new).with(partner: partner).and_return(OpenStruct.new(next_deadline: date))
```

Comment:
It's generally better to use `instance_double` instead of OpenStruct. I also tend to see this pattern (mocking the `new` method) as a code smell - it's basically the same as `allow_any_instance_of`, which can be messy. It's usually better to have the code that constructs the service in its own small method, and mock that method as returning the object.

Reply:
Thanks for the feedback. I've reworked this without any need to mock anything.